flagging_site/blueprints/flagging.py

```diff
@@ -72,13 +72,57 @@ def index() -> str:
     returns: render model on index.html
     """
     df = get_data()
-    flags = {
-        2: reach_2_model(df, rows=1)['safe'].iloc[0],
-        3: reach_3_model(df, rows=1)['safe'].iloc[0],
-        4: reach_4_model(df, rows=1)['safe'].iloc[0],
-        5: reach_5_model(df, rows=1)['safe'].iloc[0]
+    
+    # Important: Restructured these dicts for easier use.
+    # Warning - The following aren't implemented yet due to inconsistencies with http://notification.crwa.org/wqmodel/charlesflags.php (old) vs https://www.crwa.org/flagging-program.html (new):
+        # not on old CRWA page, hence not implemented below:
+        # 'Northeastern's Henderson Boathouse'
+        # 'Paddle Boston at Herter Park' -- is this the same as Reach 3's 'CRCK at Herter Park?'
+
+        # only found on old CRWA page only, hence implemented according to old website's format:
+        # 'CRCK at Herter Park'
+        # 'CRCK at Kendall Square'
+
+    homepage = {
+        2: {
+            'flag': reach_2_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_2_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'Newton Yacht Club',
+                1: 'Watertown Yacht Club',
+                2: 'Community Rowing, Inc.'
+            }
+        },
+        3: {
+            'flag': reach_3_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_3_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'CRCK at Herter Park', # Found on old CRWA page only
+                1: 'Harvard\'s Weld Boathouse' 
+            }
+        },
+        4: {
+            'flag': reach_4_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_4_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'Riverside Boat Club'
+            }
+        },
+        5: {
+            'flag': reach_5_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_5_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'Charles River Yacht Club',
+                1: 'Union Boat Club',
+                2: 'Community Boating',
+                3: 'CRCK at Kendall Square', # Found on old CRWA page only
+                4: 'Paddle Boston at Kendall Square' # Assuming this is reach 5
+            }
+        }
     }
-    return render_template('index.html', flags=flags)
+
+    return render_template('index.html', homepage=homepage)
```

Comment:
Add `model_last_updated_time=model_last_updated_time`, in combination with the comment I make about the `time` keys.